state/redis/redis.go

```diff
@@ -68,28 +69,28 @@ func (r *StateStore) Init(metadata state.Metadata) error {
 		return err
 	}
 
-	ctx := context.Background()
-	opts := redisconn.Opts{
-		DB:       0,
-		Password: redisCreds.Password,
-	}
-	conn, err := redisconn.Connect(ctx, redisCreds.Host, opts)
+	var tlsConfig *tls.Config
```

Comment:
read enableTLS from metadata and then use it here. Users would be able to set it in their components yaml file if they are talking to a secured redis endpoint with ssl.
Please see the usage in redis pubsub component:
https://github.com/dapr/components-contrib/blob/master/pubsub/redis/redis.go#L85

Reply:
+1 on this, also `tls.Config` will probably not unmarshall. 

To enable TLS if the user specified it on the YAML, its probably:

```
&tls.Config{
	InsecureSkipVerify: true,
}
```

As reported [here](https://github.com/go-redis/redis/issues/756).

Reply:
Thanks for the pointer, changed it now.